chain/client/src/client_actor.rs

```diff
@@ -1138,8 +1142,16 @@ impl ClientActor {
                         self.process_accepted_blocks(
                             accepted_blocks.write().unwrap().drain(..).collect(),
                         );
-                        for missing_chunks in blocks_missing_chunks.write().unwrap().drain(..) {
-                            self.client.shards_mgr.request_chunks(missing_chunks).unwrap();
+
+                        let request_result = self.client.shards_mgr.request_chunks(
+                            blocks_missing_chunks
+                                .write()
+                                .unwrap()
+                                .drain(..)
+                                .flat_map(|missing_chunks| missing_chunks.into_iter()),
+                        );
+                        if let Err(err) = request_result {
```

Comment:
I would also `debug_assert` here. It should not be possible for the call to `request_chunks` here to fail as far as I can tell (assuming the only way for `request_chunks` to fail is epoch out of bounds, and that we only add blocks to the `blocks_with_missing_chunks` if their epoch is known)

Reply:
It can also fail if there are problems with the DB (not able to read/write at all, missing records related to blocks, or data corruption resulting in deserialization failure). Obviously such issues are also not expected either and could be indicative of larger problems. I'm ok with a `debug_assert`, but I think the original code was dangerous because it could crash a production node due to the `unwrap` calls, whereas I would assume we want production nodes to be as resilient as possible.